db/compaction/compaction.cc

```diff
@@ -255,8 +262,15 @@ Compaction::Compaction(
                   _blob_garbage_collection_age_cutoff > 1
               ? mutable_cf_options()->blob_garbage_collection_age_cutoff
               : _blob_garbage_collection_age_cutoff),
-      penultimate_level_(EvaluatePenultimateLevel(
-          vstorage, immutable_options_, start_level_, output_level_)) {
+      penultimate_level_(
+          // For simplicity, we don't support the concept of "penultimate level"
+          // with `CompactionReason::kExternalSstIngestion` and
+          // `CompactionReason::kRefitLevel`
+          _compaction_reason == CompactionReason::kExternalSstIngestion ||
+                  _compaction_reason == CompactionReason::kRefitLevel
+              ? kInvalidLevel
```

Comment:
[Note] Excluded these two for the concept of `penultimate_level_` to avoid failure in https://github.com/facebook/rocksdb/blob/7.8.fb/db/compaction/compaction_picker.cc#L1138, where input of refit level and file ingestion overlap with the second to last level, which is fine for refit level and ingestion